src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -821,12 +1079,11 @@ private static OkObjectResult OkMutationResponse(JsonElement jsonResult)
         /// from the JsonElement representing the entity.
         /// </summary>
         /// <param name="context">RestRequestContext</param>
-        /// <param name="entity">A Json element representing one instance of the entity.</param>
+        /// <param name="dbOperationResultSet">Result set from the insert/upsert operation</param>
         /// <param name="sqlMetadataProvider">Metadataprovider for db on which to perform operation.</param>
-        /// <remarks> This function expects the Json element entity to contain all the properties
-        /// that make up the primary keys.</remarks>
+        /// <remarks> When one or more primary keys are not present an empty string will be returned.</remarks>
         /// <returns>the primary key route e.g. /id/1/partition/2 where id and partition are primary keys.</returns>
-        public static string ConstructPrimaryKeyRoute(RestRequestContext context, Dictionary<string, object?> entity, ISqlMetadataProvider sqlMetadataProvider)
+        public static string ConstructPrimaryKeyRoute(RestRequestContext context, Dictionary<string, object?> dbOperationResultSet, ISqlMetadataProvider sqlMetadataProvider)
```

Comment:
Should be a dbOperationResultSetRow

Reply:
The intention behind the name was to just make is easy to identify that the results from a successful database operation is being used to compute the primary key route. I think it should be okay if this variable name doesn't match with the exact variable name (from the caller of this function) that holds the results from the database. Considering this, I'm leaving the variable name as is

Reply:
What you are using here to generate the PK route is not a result set, but a resut set row. This distinction will become even more important when we support batch mutations.

Reply:
Updated the variable name.